src/VisualStudio/Core/Def/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -282,20 +285,23 @@ private void AttachToDocument(string moniker, ITextBuffer textBuffer)
     /// </summary>
     private ProjectInfo CreateProjectInfoForDocument(string filePath)
     {
+        // Potential calculation of _metadataReferences requires being on the main thread
+        _threadingContext.ThrowIfNotOnUIThread();
```

Comment:
Same as above: consider adding a comment here to wahtever bug we have tracking cleaning this up.